### ISSUE TITLE
parents property broken after copying analogsignalarray

### ISSUE THREAD

**Issue by JuliaSprenger** [NeuralEnsemble/python-neo#220]
After copying or time_slicing an analogsignalarray accessing the parents of the analogsignal results in an AttributeError 'AnalogSignalArray' object has no attribute 'segment'. The result is the same for a simple copy or a deep copy.

This can be seen when running:

``` python
import sys
import copy
import quantities as pq
import neo

anasigarray = neo.AnalogSignalArray(range(10)*pq.V,t_start=0*pq.s,sampling_rate=1*pq.Hz)
print 'Original: %s'%(str(anasigarray.parents))

seg = neo.Segment(name='seg1')
seg.analogsignalarrays.append(anasigarray)
print 'Appended: %s'%(str(anasigarray.parents))

anasigarray2 = anasigarray.time_slice(None,5*pq.s)
try:
    print 'After slicing: %s'%(str(anasigarray2.parents))
except AttributeError as detail:
    print 'After slicing: AttributeError %s'%detail

anasigarray3 = copy.deepcopy(anasigarray)
try:
    print 'After deep copy: %s'%(str(anasigarray3.parents))
except AttributeError as detail:
    print 'After deep copy: AttributeError %s'%detail

anasigarray4 = anasigarray.copy()
try:
    print 'After copy: %s'%(str(anasigarray4.parents))
except AttributeError as detail:
    print 'After copy: AttributeError %s'%detail
```

This code prints:

> Original: (None, None)
> Appended: (None, None)
> After slicing: AttributeError 'AnalogSignalArray' object has no attribute 'segment'
> After deep copy: AttributeError 'AnalogSignalArray' object has no attribute 'segment'
> After copy: AttributeError 'AnalogSignalArray' object has no attribute 'segment'

In the case of time_slice this can be fixed by additionally copying the parents of the old analogsignalarray to the newly generated analogsignalarray in the time_slice function. But this does not solve the same issue when using copy.

**Comment by mczerwinski** [comment]
I was using deepcopy in my code.
When copying analogsignals it worked ok.
On the segment it had problems, probably with recursive implementation of copy/deepcopy - the returned copies were simplified/derived of some parameters (annotations, quantities)

**Comment by samuelgarcia** [comment]
Fixed by #399.
